adamacs/ingest/session.py

```diff
@@ -1,101 +1,126 @@
-from cgitb import scanvars
-import pathlib
-import csv
-from datetime import datetime
 import re
-from element_session import session_with_id
+import pathlib
+import warnings
 from itertools import groupby
+from ..pipeline import subject, session, scan
+from ..paths import get_imaging_root_data_dir
 from datajoint.errors import DuplicateError
-import warnings
-from element_calcium_imaging import scan
+from element_interface.utils import find_full_path
+
+""" Notes from Chris
+1. I placed previous default (r'C:\\datajoint') as the default for
+    get_imaging_root_data_dir when the relevant config field is empty
+2. For improved functionality, we might make better use of get_scan_image_files, which
+    concatenates the root and session dirs. Currently, the function below has to find
+    the session directory from the root, rather than being root agnostic.
+"""
 
 
-def ingest_session_scan(session_key, session_path=r'C:\datajoint', verbose=False):
+def ingest_session_scan(session_key, root_paths=get_imaging_root_data_dir(),
+                        verbose=False):
     """Locate all directories in session_path that are part
     of the sesssion. Extract the following attributes from
     the directory names:
-    Subject
-    session_datetime
-    User
+        Subject
+        session_datetime
+        User
     """
+    if not verbose:
+        warnings.filterwarnings('ignore')
 
-    p = pathlib.Path(session_path)
-
-    session_pattern = session_key  # Session key can be exactly matched
-    scan_pattern = "scan.{8}"
-
-    dirs = [x for x in p.iterdir() if x.is_dir() and session_key in x.name]
+    paths = [pathlib.Path(path) for path in root_paths]
+    valid_paths = [p for p in paths if p.is_dir()]
+    match_paths = []
+    for p in valid_paths:
+        match_paths.extend(list(p.rglob(f'*{session_key}*')))
     
-    n_scans = len(dirs)
-    if verbose: print(f"Number of scans found: {n_scans}")
+    n_scans = len(match_paths)
+    if verbose:
+        print(f'Number of scans found: {n_scans}')
 
-    basenames = [x.name for x in dirs]
-    
+    scan_pattern = "scan.{8}"
+    basenames = [x.name for x in match_paths]
     scan_keys = [re.findall(scan_pattern, x) for x in basenames]
```

Comment:
`re.findall` returns a list, resulting `scan_keys` being a list of list, is this intended? perhaps using `re.search` is more suitable?

Reply:
Daniel's draft here captured each data type as a list, checked that that list only contained one unique value and, if so, proceeded with data insertion. This is not how I would have designed it from the ground up, but I conformed to the format thinking that there might be more reasons to do so than I currently understand.